src/controllers/midi/portmidienumerator.cpp

```diff
@@ -20,44 +20,33 @@ bool recognizeDevice(const PmDeviceInfo& deviceInfo) {
                      .startsWith(kMidiThroughPortPrefix, Qt::CaseInsensitive);
 }
 
-} // namespace
-
-PortMidiEnumerator::PortMidiEnumerator() {
-    PmError err = Pm_Initialize();
-    // Based on reading the source, it's not possible for this to fail.
-    if (err != pmNoError) {
-        qWarning() << "PortMidi error:" << Pm_GetErrorText(err);
-    }
-}
-
-PortMidiEnumerator::~PortMidiEnumerator() {
-    qDebug() << "Deleting PortMIDI devices...";
-    QListIterator<Controller*> dev_it(m_devices);
-    while (dev_it.hasNext()) {
-        delete dev_it.next();
-    }
-    PmError err = Pm_Terminate();
-    // Based on reading the source, it's not possible for this to fail.
-    if (err != pmNoError) {
-        qWarning() << "PortMidi error:" << Pm_GetErrorText(err);
-    }
-}
-
-bool namesMatchMidiPattern(const QString& input_name,
-        const QString& output_name) {
-    // Some platforms format MIDI device names as "deviceName MIDI ###" where
-    // ### is the instance # of the device. Therefore we want to link two
-    // devices that have an equivalent "deviceName" and ### section.
-    QRegExp deviceNamePattern("^(.*) MIDI (\\d+)( .*)?$");
-
-    int inputMatch = deviceNamePattern.indexIn(input_name);
-    if (inputMatch == 0) {
-        QString inputDeviceName = deviceNamePattern.cap(1);
-        QString inputDeviceIndex = deviceNamePattern.cap(2);
-        int outputMatch = deviceNamePattern.indexIn(output_name);
-        if (outputMatch == 0) {
-            QString outputDeviceName = deviceNamePattern.cap(1);
-            QString outputDeviceIndex = deviceNamePattern.cap(2);
+// Some platforms format MIDI device names as "deviceName MIDI ###" where
+// ### is the instance # of the device. Therefore we want to link two
+// devices that have an equivalent "deviceName" and ### section.
+const QRegularExpression midiDeviceNameRegex("^(.*) MIDI (\\d+)( .*)?$");
+
+const QRegularExpression inputRegex("^(.*) in (\\d+)( .*)?$");
```

Comment:
Should work, is more efficient, and easier to read:

`const QRegularExpression kInputRegex = QStringLiteral("^(.*) in (\\d+)( .*)?$");`

Also the naming of constants doesn't comply with our coding style.